packages/component-library/src/components/plmg-slider/plmg-slider.tsx

```diff
@@ -0,0 +1,314 @@
+    // Ignore Empty Strings
+    if (ev.target.value == '') return;
+    // Get an array of allowed inputs
+    const SET_ALLOWED_INPUTS = this.getAllowedInputs();
+    // Check input is allowed value, within range and one of the allowed inputs
+    const newValue = parseFloat(ev.target.value);
+    if (
+      isNaN(newValue) ||
+      newValue > this.max ||
+      newValue < this.min ||
+      !SET_ALLOWED_INPUTS.includes(newValue)
+    )
+      return;
+    this.updateValue(newValue);
+  }
+
+  private setValues() {
+    if (!this.rangeValues) return;
+    this.min = this.rangeValues[0];
+    this.max = this.rangeValues[this.rangeValues.length - 1];
+    if (this.defaultValue >= this.min && this.defaultValue <= this.max) {
+      this.updateValue(this.defaultValue);
+    } else {
+      this.updateValue(this.min);
+    }
+    if (!this.step) {
+      this.stepValue = (this.max - this.min) / 100;
+    } else {
+      this.stepValue = this.step;
+    }
+  }
+
+  private resizeHandler() {
+    this.trackWidth = this.ref.getBoundingClientRect().width;
+  }
+
+  connectedCallback() {
+    this.setValues();
+  }
+
+  componentDidLoad() {
+    this.trackWidth = this.ref.getBoundingClientRect().width;
+    this.abortResizeListener = new AbortController();
+    window.addEventListener('resize', () => this.resizeHandler(), {
+      signal: this.abortResizeListener.signal,
+    });
+  }
+
+  disconnectedCallback() {
+    if (this.abortResizeListener) this.abortResizeListener.abort();
+  }
+  render() {
+    const containerClasses = {
+      'plmg-slider-container': true,
+      marks: this.marks,
+    };
+
+    return (
+      <Host value={this.value}>
+        <div class={'plmg-component-container'}>
+          <div
+            class={containerClasses}
+            ref={(el) => (this.ref = el as HTMLDivElement)}
+          >
+            {this.thumbLabel && (
+              <div class={'thumb-label-container'}>
+                <output
+                  style={this.setThumbPosition()}
+                  class={'plmg-slider-thumb-label'}
+                  htmlFor={this.name}
+                >
+                  {this.value}
+                  <span class={'plmg-thumb-triangle'} />
+                </output>
+              </div>
+            )}
+
+            <input
+              min={this.min}
+              max={this.max}
+              name={this.name}
+              value={this.value}
+              aria-valuemin={this.min}
+              aria-valuemax={this.max}
+              aria-valuenow={this.value}
+              step={this.stepValue}
+              onInput={(ev) => this.handleSliderChange(ev)}
+              style={{ background: this.setBackgroundProgressFill() }}
+              id="range"
+              type="range"
+            />
+
+            {this.marks && (
+              <div class={'labels'}>
+                {this.rangeValues.map((item, index) => (
+                  <div
+                    class={'label'}
```

Comment:
Prefix class with `plmg-`.
As above, should the class be called `mark` since that relates to the name of the prop?

Reply:
yes. fixed.